packages/base/src/api/client.ts

```diff
@@ -18,21 +18,25 @@ export abstract class BaseApiClient {
     this.apiSecret = params.apiSecret;
   }
 
-  protected async init(): Promise<void> {
-    const userPass = { Username: this.apiKey, Password: this.apiSecret };
-    const poolData = { UserPoolId: this.getPoolId(), ClientId: this.getPoolClientId() };
-    this.api = await createAuthenticatedApi(userPass, poolData, this.getApiUrl());
+  protected async init(): Promise<AxiosInstance> {
+    if (!this.api) {
+      const userPass = { Username: this.apiKey, Password: this.apiSecret };
+      const poolData = { UserPoolId: this.getPoolId(), ClientId: this.getPoolClientId() };
+      this.api = createAuthenticatedApi(userPass, poolData, this.getApiUrl());
```

Comment:
should we await here? (instead of promisifying the type bellow?)

Reply:
Nope! The point of assigning a promise directly is to avoid more than one initialization in parallel. Otherwise, two simultaneous calls on the same uninitialized client instance would trigger two calls to `createAuthenticatedApi`.